File: js-time-trial/middle-earth.test.js
import {
  reverseString,
  countVowels,
  isPalindrome,
  capitalizeWords,
  replaceSpacesWithUnderscores,
  removeVowels,
  sortCharacters,
} from "./middle-earth";
describe("Middle Earth", () => {
  test("reverseString", () => {
    expect(reverseString("Frodo")).toBe("odorF");
    expect(reverseString("Gollum")).toBe("mulloG");
    expect(reverseString("")).toBe("");
  });

  test("countVowels", () => {
    expect(countVowels("Frodo")).toBe(2);
    expect(countVowels("Gollum")).toBe(2);
    expect(countVowels("aeiou")).toBe(5);
  });

  test("isPalindrome", () => {
    expect(isPalindrome("sagas")).toBe(true);
    expect(isPalindrome("Frodo")).toBe(false);
    expect(isPalindrome("")).toBe(true);
  });

  test("capitalizeWords", () => {
    expect(capitalizeWords("frodo baggins")).toBe("Frodo Baggins");
    expect(capitalizeWords("samwise gamgee")).toBe("Samwise Gamgee");
    expect(capitalizeWords("middle earth")).toBe("Middle Earth");
  });

  test("replaceSpacesWithUnderscores", () => {
    expect(replaceSpacesWithUnderscores("frodo baggins")).toBe("frodo_baggins");
    expect(replaceSpacesWithUnderscores("samwise gamgee")).toBe(
      "samwise_gamgee"
    );
    expect(replaceSpacesWithUnderscores("middle earth")).toBe("middle_earth");
  });

  test("removeVowels", () => {
    expect(removeVowels("Frodo")).toBe("Frd");
    expect(removeVowels("Gollum")).toBe("Gllm");
    expect(removeVowels("aeiou")).toBe("");
  });

  test("sortCharacters", () => {
    expect(sortCharacters("Frodo")).toBe("Fdoor");
    expect(sortCharacters("Gollum")).toBe("Gllmou");
    expect(sortCharacters("Middle Earth")).toBe("EMaddehilrt");
  });
});

File: java-time-trial/src/test/java/com/javatimetrial/app/MiddleEarthTest.java
package com.javatimetrial.app;

import org.junit.Test;
import static org.junit.Assert.*;
public class MiddleEarthTest {
    @Test
    public void testReverseString() {
        assertEquals("odorF", MiddleEarth.reverseString("Frodo"));
        assertEquals("mulloG", MiddleEarth.reverseString("Gollum"));
        assertEquals("", MiddleEarth.reverseString(""));
    }

    @Test
    public void testCountVowels() {
        assertEquals(2, MiddleEarth.countVowels("Frodo"));
        assertEquals(2, MiddleEarth.countVowels("Gollum"));
        assertEquals(5, MiddleEarth.countVowels("aeiou"));
    }

    @Test
    public void testIsPalindrome() {
        assertTrue(MiddleEarth.isPalindrome("sagas"));
        assertFalse(MiddleEarth.isPalindrome("Frodo"));
        assertTrue(MiddleEarth.isPalindrome(""));
    }

    @Test
    public void testCapitalizeWords() {
        assertEquals("Frodo Baggins", MiddleEarth.capitalizeWords("frodo baggins"));
        assertEquals("Samwise Gamgee", MiddleEarth.capitalizeWords("samwise gamgee"));
        assertEquals("Middle Earth", MiddleEarth.capitalizeWords("middle earth"));
    }

    @Test
    public void testReplaceSpacesWithUnderscores() {
        assertEquals("frodo_baggins", MiddleEarth.replaceSpacesWithUnderscores("frodo baggins"));
        assertEquals("samwise_gamgee", MiddleEarth.replaceSpacesWithUnderscores("samwise gamgee"));
        assertEquals("middle_earth", MiddleEarth.replaceSpacesWithUnderscores("middle earth"));
    }

    @Test
    public void testRemoveVowels() {
        assertEquals("Frd", MiddleEarth.removeVowels("Frodo"));
        assertEquals("Gllm", MiddleEarth.removeVowels("Gollum"));
        assertEquals("", MiddleEarth.removeVowels("aeiou"));
    }

    @Test
    public void testSortCharacters() {
        assertEquals("Fdoor", MiddleEarth.sortCharacters("Frodo"));
        assertEquals("Gllmou", MiddleEarth.sortCharacters("Gollum"));
        assertEquals("EMaddehilrt", MiddleEarth.sortCharacters("Middle Earth"));
    }
}
